lib/templates.js

```diff
@@ -173,31 +177,40 @@ const createAPIResourceTemplate = (serviceName, serviceDefinition, appConfig, pa
 
             ...ordExtensions,
         };
-
-        apiResources.push(obj);
+        // visibility can also be set as an ORD extension; in this case, it overwrites the default visibility
+        if (obj.visibility === RESOURCE_VISIBILITY.public) apiResources.push(obj);
     });
 
-    if (apiResources.length > 0) return apiResources;
+    return apiResources;
 };
 
-const createEventResourceTemplate = (serviceName, eventDefinition, appConfig, packageIds) => {
-    const ordExtensions = readORDExtensions(eventDefinition);
-    return {
+/**
+ * This is a template function to create Event Resource object for Event Resource Array.
+ * There can be only one event resource per service because all events are using the same protocol, they are always Cloud Events.
+ *
+ * @param {string} serviceName The name of the service.
+ * @param {object} serviceDefinition The definition of the service
+ * @returns {Array} An single-item array of objects for the Event Resources.
+ */
+const createEventResourceTemplate = (serviceName, serviceDefinition, appConfig, packageIds) => {
+    const ordExtensions = readORDExtensions(serviceDefinition);
+    if (!!ordExtensions.visibility && ordExtensions.visibility !== RESOURCE_VISIBILITY.public) return [];
+    return [{
         ordId: `${appConfig.ordNamespace}:eventResource:${serviceName}:v1`,
         title:
-            eventDefinition["@title"] ??
-            eventDefinition["@Common.Label"] ??
+            serviceDefinition["@title"] ??
+            serviceDefinition["@Common.Label"] ??
             `ODM ${appConfig.appName.replace(/[^a-zA-Z0-9]/g, "")} Events`,
         shortDescription: `${serviceName} event resource`,
         description:
-            eventDefinition['@description'] ?? eventDefinition['@Core.Description'] ??
+            serviceDefinition['@description'] ?? serviceDefinition['@Core.Description'] ??
             "CAP Event resource describing events / messages.",
         version: "1.0.0",
         lastUpdate: appConfig.lastUpdate,
         releaseStatus: "active",
-        partOfPackage: _getPackageID(appConfig.ordNamespace, packageIds, 'event'),
+        partOfPackage: _getPackageID(appConfig.ordNamespace, packageIds, ORD_RESOURCE_TYPE.event),
         partOfGroups: [_getGroupID(serviceName, defaults.groupTypeId, appConfig)],
-        visibility: "public",
+        visibility: RESOURCE_VISIBILITY.public,//default value that can be overwritten by ORD extension
```

Comment:
The comment is not needed

Reply:
Everything here can be overwritten by ORDExtensions I assume? Maybe that comment should be moved to the "higher level"

Reply:
Moved to the function doc comment.